docs/modules/SDDs/pages/0030-argocd-multitenancy.adoc

```diff
@@ -0,0 +1,101 @@
+
+With this approach, the team responsible for the cluster itself can continue using the current ArgoCD project (`syn`) and root application (`root`).
+If there's no applications managed by another team, this approach allows the team owning the cluster to continue working as before.
+
+However, if another team wants to deploy applications on the cluster through Project Syn, an additional ArgoCD project and root application can be bootstrapped.
+Each team's root application is managed independently from the default root application (`root`).
+
+The bootstrap process for the additional root applications is initiated by defining the team in the configuration hierarchy in parameter `syn_teams`.
+This parameter is not associated with a specific component and becomes a reserved parameter name which components can't use.
+
+To ensure components can be assigned to teams, the argocd component library needs to be made "team-aware".
+For each team, the list of component instances owned by that team is specified in `syn_teams.<team-name>.instances`.
+This list must always be processed with the `renderArray()` Jsonnet function (or an equivalent).
+The https://github.com/projectsyn/component-argocd/blob/1227b46c2db6e6959ce75aa7ab44db975f33bb3e/lib/argocd.libjsonnet[argocd component library] can read this list to determine whether the instance which is being rendered should be assigned to a team's root application or the default root application.
+If a component instance is listed in `instances` of multiple teams a compilation error is raised.
+Any component instance which isn't listed in `instances` of any team is assigned to the default root application.
+
+To ensure each team's applications are independent, each team's ArgoCD application manifests (including the team's root application) are stored in path `apps-<team-name>/` in the cluster catalog repository.
+To do so, each component's `kapitan.compile` entry for the ArgoCD application needs to be adjusted to write the application manifest into a well-defined path matching the owning team's ArgoCD project.
+This change can be implemented globally in the component template.
+By implementing the dynamic apps path in the catalog in the component template, we don't need to adjust Commodore's logic for copying the output of `kapitan compile` into the cluster catalog.
```

Comment:
This approach sounds like it will make the migration much simpler. I'd still want to update the component template to generate multi-tenant capable components by default, but having an explicit flag will make migrating existing components easier (especially if the template update doesn't apply cleanly to existing components).

I'll update the text accordingly.